core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -140,16 +140,19 @@ class Partition(val topic: String,
 
   def delete() {
     // need to hold the lock to prevent appendMessagesToLeader() from hitting I/O exceptions due to log being deleted
+    val start = time.milliseconds
     inWriteLock(leaderIsrUpdateLock) {
       assignedReplicaMap.clear()
       inSyncReplicas = Set.empty[Replica]
       leaderReplicaIdOpt = None
+      val topicPartition = TopicAndPartition(topic, partitionId)
       try {
-        logManager.deleteLog(TopicAndPartition(topic, partitionId))
+        val renamedDir = logManager.asyncDelete(topicPartition)
         removePartitionMetrics()
+        info(s"Log for $topicPartition renamed to $renamedDir and scheduled for deletion. Spent ${time.milliseconds - start} ms.")
```

Comment:
The log line is slightly weird. Not sure how useful it is to have a "spent" part given that it can be derived from logging timestamps. It is also probably redundant to the log that is already inside `asyncDelete`. That said, if you want this I would go with `... renamed to $renamedDir in $time milliseconds and scheduled for deletion.`